xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java

```diff
@@ -204,7 +215,12 @@ void verifySubjectAltNameInChain(X509Certificate[] peerCertChain) throws Certifi
   @Override
   public void checkClientTrusted(X509Certificate[] chain, String authType, Socket socket)
       throws CertificateException {
-    delegate.checkClientTrusted(chain, authType, socket);
+    if (delegates != null) {
+      String trustDomain = SpiffeUtil.extractSpiffeId(chain).get().getTrustDomain();
+      delegates.get(trustDomain).checkClientTrusted(chain, authType, socket);
```

Comment:
If "trustDomain" is a not key of the map, this will throw an IllegalArgumentException, I think. So that the calling code can handle in the same way it would any validation failure, can we check if the key is in the map and if not throw a CertificateException instead?

Similarly below.